src/model.cpp

```diff
@@ -547,4 +547,72 @@ bool Model::fixVariableInterfaces()
     return allOk;
 }
 
+static const size_t MAX_SIZE_T = std::numeric_limits<size_t>::max();
+
+bool isComponentEmpty(const ComponentPtr &component)
+{
+    for (size_t i = component->componentCount() - 1; i != MAX_SIZE_T; --i) {
+        if (isComponentEmpty(component->component(i))) {
+            component->removeComponent(i);
+        }
+    }
+
+    return (component->variableCount() + component->resetCount() + component->componentCount()) == 0
+           && component->math().empty()
+           && !component->isImport();
+}
+
+void checkUnits(const ComponentPtr &component, std::vector<UnitsPtr> &unused)
+{
+    for (size_t v = 0; v < component->variableCount(); ++v) {
+        auto u = component->variable(v)->units();
+        if (u != nullptr) {
+            auto it = std::find_if(unused.begin(), unused.end(), [=](const UnitsPtr &t) -> bool {
+                // KRM Not sure whether including names is a good idea or not.  They should be linked by
+                //     pointers only anyway, but the findUnits function is still testing names and equivalence ...
+                return t == u || u->name() == t->name();
```

Comment:
Replace with `return (t == u) || (u->name() == t->name()); `.

Reply:
f15e055d

Reply:
Removed this function in 8837c78c